### PR TITLE
Add client support for getting ticket metrics...

### PR DESCRIPTION
**Pull request by jnormington**
Using the current search filter ability via the search api. Then utilise the zendesk side loading feature on the tickets endpoint

For issue #6

### CODE REVIEW

File: conf/metrics.go [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+func (m MetricOption) Valid() error {
+	var match bool
+
+	if reflect.DeepEqual(m, MetricOption{}) {
+		return errEmptyMetricOptions
+	}
+
+	for _, vma := range validMetricAttributes {
+		if vma == m.Attribute {
+			match = true
```
Comment (achilleasa) [inline conf/metrics.go]: break?

File: conf/metrics.go [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+	}
+
+	if !match {
+		return errInvalidAttribute
+	}
+
+	match = false
+	for _, vmu := range validMetricUnits {
+		if vmu == m.Unit {
+			match = true
```
Comment (achilleasa) [inline conf/metrics.go]: break?

File: conf/metrics.go [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+
+	return -1
+}
+
+// Valid return either true or false validating options required for metric reports.
+// It doesn't validate against the grouping as not all metric reports require it.
+func (m MetricOption) Valid() error {
+	var match bool
+
+	if reflect.DeepEqual(m, MetricOption{}) {
```
Comment (achilleasa) [inline conf/metrics.go]: I would recommend against using reflection in non-test code. How about a `IsEmpty()/IsValid()` method on `MetricOption`?

File: conf/metrics.go [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+		FirstResolutionTime,
+		FullResolutionTime,
+		AgentWaitTime,
+		RequesterWaitTime,
+		OnHoldTime,
+	}
+
+	validMetricUnits = [2]MetricSubMetric{BusinessMetric, CalendarMetric}
+
+	errInvalidAttribute   = fmt.Errorf("The metric attribute is not valid must be one of %s", validMetricAttributes)
```
Comment (achilleasa) [inline conf/metrics.go]: You could always export these so the caller can do explicit error matching if they need to for their particular application

File: conf/metrics.go (hunk 1 above)
Comment (jnormington) [inline conf/metrics.go]: Go catch thanks Achilleas

File: conf/metrics.go (hunk 3 above)
Comment (jnormington) [inline conf/metrics.go]: Thanks for the tip. Will implement the is empty method on metric option.

File: conf/metrics.go (hunk 2 above)
Comment (jnormington) [inline conf/metrics.go]: Good catch added now. Thanks

**Pull request merged**